tensorflow/python/ops/math_grad.py

```diff
@@ -1308,6 +1308,38 @@ def _SelectGrad(op, grad):
   return (None, array_ops.where(c, grad, zeros), array_ops.where(
       c, zeros, grad))
 
+@ops.RegisterGradient("SelectV2")
+def _SelectGradV2(op, grad):
+  c = op.inputs[0]
+  x = op.inputs[1]
+  y = op.inputs[2]
+  zeros = array_ops.zeros([], dtype=grad.dtype.base_dtype)
+  gx = array_ops.where_v2(c, grad, zeros)
+  gx_shape = array_ops.shape(gx)
+  x_shape = array_ops.shape(x)
+  rankdiff_x = array_ops.rank(gx) - array_ops.rank(x)
+  # Reduce away broadcasted leading dims.
+  gx = math_ops.reduce_sum(gx, axis=math_ops.range(rankdiff_x))
+  # Reduce but keep x's 1-valued dims which were broadcast.
+  axis = array_ops.where(gx_shape[rankdiff_x:] > x_shape)
+  # tf.where returns 2D so squeeze.
+  axis = array_ops.squeeze(axis)
+  gx = math_ops.reduce_sum(gx, keepdims=True, axis=axis)
+
+  gy = array_ops.where_v2(c, zeros, grad)
+  gy_shape = array_ops.shape(gy)
+  y_shape = array_ops.shape(y)
+  rankdiff_y = array_ops.rank(gy) - array_ops.rank(y)
+  # Reduce away broadcasted leading dims.
+  gy = math_ops.reduce_sum(gy, axis=math_ops.range(rankdiff_y))
+  # Reduce but keep y's 1-valued dims which were broadcast.
+  axis = array_ops.where(gy_shape[rankdiff_y:] > y_shape)
```

Comment:
Why is this where and not where_v2?

Reply:
Thanks @alextp. The PR has been updated to use where_v2. In both cases (where_v2 and where), they behave the same though as `else` and `then` are `None` so follow the same rule. Updated to use `where_v2` for consistency.